Remotion/Web/Development.WebTesting/HostingStrategies/Configuration/TestSiteLayoutConfiguration.cs

```diff
@@ -0,0 +1,39 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Remotion.Web.Development.WebTesting.Configuration;
+
+namespace Remotion.Web.Development.WebTesting.HostingStrategies.Configuration
+{
+  /// <inheritdoc />
+  public class TestSiteLayoutConfiguration : ITestSiteLayoutConfiguration
+  {
+    public TestSiteLayoutConfiguration (WebTestConfigurationSection configSettings)
+    {
+      RootPath = configSettings.TestSiteLayoutConfiguration.RootPath;

```

Comment:
Arg-check. there's some inconcsistency with the argument checks. please check all new methods in this PR for correct argumetn checks.

Sepearate issue: we need to check the existing webtest-codebase to make sure everything public or protected has argument checks before we go and implement nullable reference types. Otherwise, we have to fix all those false-nullable parameters.